lib/integrations/bot_processor_service.rb

```diff
@@ -7,7 +7,7 @@ def perform
 
```

Comment:
@sojan-official `:hook!` attributes shouldn't it match `:agent_bot!` in the child class in the initializer, changing the sequence here. Because seems hook is always nil in this case.

Reply:
@tejaswinichile  it wouldn't match `hook` . the gem defines a new setter and getter methods named `agent_bot` .
https://github.com/barsoom/attr_extras/blob/4913a40d2cea9cd90fbdc444441e35268cf3c6c3/lib/attr_extras/attr_initialize.rb#L23

So in case of CMSL processor service hook will be nill.  ( we might have to look into the naming ) . but for now we could do  `(hook&.account || agent_bot&.account) `

also add a comment mentioning that in CSML processor service, the argument is agent bot. 

or we could just rename it to `bot_hook` or `bot_object` something and make this consistent across the base classes and inherited classes.

Reply:
> Doing this for quick fix (hook&.account || agent_bot&.account) 

> or we could just rename it to bot_hook or bot_object something and make this consistent across the base classes and inherited classes.

<< I will take this one as a later ticket.